### PR TITLE
Story/cico 5

### PR DESCRIPTION
**Pull request by ajay3568yadav**
removed additional files and cleared up the cells

### CODE REVIEW

File: csvGenerator.ipynb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "2c0597b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "auth_object = authObject()\n",
+    "auth_object.authType  = 'oauth'\n",
+    "auth_object.access_token=\"b7f4880b-7e33-4eca-b5a5-29c73adce87a\"\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: should never be pushed!!!! this is secret.

File: csvGenerator.ipynb [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+   "execution_count": 3,
+   "id": "37167759",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def downloadFile(file_id, citesphere_token):\n",
+    "    giles_url = f\"https://diging.asu.edu/geco-giles-staging/api/v2/resources/files/{file_id}/content\"\n",
+    "    folder_path = \"citesphere-connector\"\n",
+    "    os.makedirs(folder_path, exist_ok=True)\n",
+    "    filename = os.path.join(folder_path, f\"{file_id}.pdf\")\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: files are not necessarily pdfs

File: csvGenerator.ipynb [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    "    response = requests.get(giles_url, headers=headers)\n",
+    "    if response.status_code == 200:\n",
+    "        with open(filename, \"wb\") as file:\n",
+    "            file.write(response.content)\n",
+    "        return filename  \n",
+    "    else:\n",
+    "        return None  \n",
+    "\n",
+    "def addToCSV(itm:list,csv_dict:dict):\n",
+    "    for value in itm[\"items\"]:\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: if `itm` is a list, the name should indicate that. also it's misspelled (should be "item" not "itm")

File: csvGenerator.ipynb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    "    }\n",
+    "    response = requests.get(giles_url, headers=headers)\n",
+    "    if response.status_code == 200:\n",
+    "        with open(filename, \"wb\") as file:\n",
+    "            file.write(response.content)\n",
+    "        return filename  \n",
+    "    else:\n",
+    "        return None  \n",
+    "\n",
+    "def addToCSV(itm:list,csv_dict:dict):\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: if you use type hints, that should be used throughout, not just in one function and there should be blank before the type (`itm: list`).

File: csvGenerator.ipynb [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    "groups=connector.get_groups()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "37167759",
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (jdamerow) [inline csvGenerator.ipynb]: functions need documentation

File: csvGenerator.ipynb (hunk 3 above)
Comment (jdamerow) [inline csvGenerator.ipynb]: also, what is `value`? needs a better variable name.

File: csvGenerator.ipynb [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+   "execution_count": 2,
+   "id": "2c0597b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "auth_object = authObject()\n",
+    "auth_object.authType  = 'oauth'\n",
+    "auth_object.access_token=\"b7f4880b-7e33-4eca-b5a5-29c73adce87a\"\n",
+    "connector = CitesphereConnector(\"https://diging-dev.asu.edu/citesphere-review/api\", auth_object)\n",
+    "groups=connector.get_groups()"
```
Comment (jdamerow) [inline csvGenerator.ipynb]: this should be where groups are actually used

File: csvGenerator.ipynb [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+      "FILEfItViY7iUgLM\n"
+     ]
+    }
+   ],
+   "source": [
+    "csv_dict={}\n",
+    "for group in groups:\n",
+    "    groupID = group[\"id\"]\n",
+    "    collections = connector.get_collections(groupID)\n",
+    "    for x in collections[\"collections\"]:\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: come on `x`? really?

File: csvGenerator.ipynb [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+    "        numPages = math.ceil(x[\"numberOfItems\"] / 50)\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: why 50? also, needs to be a constant.

File: csvGenerator.ipynb [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    "        if numPages == 0:\n",
+    "            continue\n",
+    "\n",
+    "        if numPages == 1:\n",
+    "            itm = connector.get_collection_items_pg(groupID, x[\"key\"], 1)\n",
+    "            addToCSV(itm,csv_dict)\n",
+    "        else:\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: this if/else seems unneccesary. you can iterate over range(1)

File: csvGenerator.ipynb [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "d42eaa42",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('citesphereCSV.csv', 'w', newline='') as file:\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: metadata should be written to the csv as files are being downloaded

File: csvGenerator.ipynb [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+    "    writer = csv.writer(file)\n",
+    "    key, val = random.choice(list(csv_dict.items()))\n",
```
Comment (jdamerow) [inline csvGenerator.ipynb]: what is this for?

**Review by jdamerow — CHANGES_REQUESTED** [review]
Pull request shoudl be made to develop not main.

- strings should be put into constants at the beginning of the notebook
- Python uses snakecase, so function names need to be changed
- there are more than just the uploaded files. those need to be downloaded too.
- you need to work on your variable names

**Comment by jdamerow** [comment]
also, please reopen this pull request, do not create a new one.

File: csvGenerator.ipynb (hunk 11 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: Added Comment, The given section is for creating a CSV file

Comment (jdamerow) [reply csvGenerator.ipynb]: Do you mean this comment:
`#Create the CSV file with all the metadata and file path of the downloaded files.`
That does not explain what this line of code is doing. Why do you take a random entry form the dictionary?

File: csvGenerator.ipynb (hunk 3 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: corrected the names of variables

File: csvGenerator.ipynb (hunk 4 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: used hints in all functions

File: csvGenerator.ipynb (hunk 7 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: changed the variable name

File: csvGenerator.ipynb (hunk 8 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: stored in a constant, max_default.

File: csvGenerator.ipynb (hunk 10 above)
Comment (ajay3568yadav) [reply csvGenerator.ipynb]: fixed that.

**Comment by jdamerow** [comment]
Please reread our github guidelines and follow them.

**Comment by jdamerow** [comment]
Please change pull request to "develop" not "main".
> Pull request shoudl be made to develop not main.

**Comment by jdamerow** [comment]
Please clean the notebook output before pushing.

**Pull request closed without merging**